src/forms/text.rs

```diff
@@ -0,0 +1,421 @@
+use crossterm::event::{KeyCode, KeyEvent, KeyModifiers};
+
+use crate::{events::Event, pixel, screen::Screen};
+
+use super::{FormField, FormOptions, FormValidationResult, FormValue};
+
+/// Generic text input
+///
+/// This form field generates a generic text input, that handles keyboard input (moving cursor, backspacing / deleting, home / end)
+/// This field is inactive by default, you need to set it active once created
+///
+/// Outputs `FormValue::String`
+///
+/// see example `form-input` for basic usage
+pub struct Text {
+    screen: Screen,
+    dirty: bool,
+    active: bool,
+    input_buffer: String,
+    cursor_pos: usize,
+    options: FormOptions,
+}
+
+impl Text {
+    pub fn new(w: u32, options: FormOptions) -> Self {
+        Self {
+            screen: Screen::new(w, 1),
+            dirty: true,
+            active: false,
+            input_buffer: String::new(),
+            cursor_pos: 0,
+            options,
+        }
+    }
+
+    /// Sets a specific value inside the field
+    pub fn set_input_buffer(&mut self, input: &str) {
+        self.dirty = true;
+        self.input_buffer = String::from(input);
+        self.move_cursor(i32::MAX);
+    }
+
+    /// Clear the field
+    pub fn clear_input_buffer(&mut self) {
+        self.dirty = true;
+        self.input_buffer = String::new();
+        self.cursor_pos = 0;
+    }
+
+    /// Insert a character at the position of the cursor
+    pub fn put_char(&mut self, chr: char) {
+        let mut new_buffer = self
+            .input_buffer
+            .chars()
+            .take(self.cursor_pos)
+            .collect::<String>();
+        new_buffer.push(chr);
+        new_buffer.push_str(
+            &self
+                .input_buffer
+                .chars()
+                .skip(self.cursor_pos)
+                .collect::<String>(),
+        );
+        self.input_buffer = new_buffer;
```

Comment:
Never mind, we can just `self.input_buffer.insert(self.cursor_pos, chr)`, right?

Reply:
Actually, it uses byte offset (instead of char, as I would expect), so it can behave not as expected in some cases (or even panic when not on char boundary).

Reply:
Actually I changed thecode this way because of the byte offset instead of chars, I was getting panics when I wrote some special characters on the inputs

Reply:
Tested option 1, worked successfully, we don't even need that `.collect::<String>()`

I prefer this one as it gets straight to the point. I'm learning a lot with your suggestions 😄

Reply:
>I'm learning a lot with your suggestions

That's great to hear! I'm learning too